cmake/OpenLocoUtility.cmake

```diff
@@ -264,3 +264,92 @@ endfunction()
 function(loco_add_executable TARGET)
     _loco_add_target(${TARGET} NULL ${ARGN} EXECUTABLE)
 endfunction()
+
+# Add headers check to verify all headers carry their dependencies.
+function(loco_enable_headers_check TARGET)
```

Comment:
why not add this to the end of `_loco_add_target` then you don't need to manually add it to every library.

Reply:
Can not currently do it, we add the link libraries after the function call but the new targets need them to work, I want to rework the cmake stuff a bit but lets stick to this for now.

Reply:
Changed it, now it checks both, see if public headers include private stuff and for private headers it checks if they would compile on their own.